Code/Tools/Libs/GuiFoundation/Action/Implementation/ActionMap.cpp

```diff
@@ -266,3 +297,53 @@ const ezTreeNode<ezActionMapDescriptor>* ezActionMap::GetChildByName(const ezTre
   }
   return nullptr;
 }
+
+const ezActionMap::TreeNode* ezActionMap::BuildActionTree()
+{
+  ezUInt32 uiCurrentTransitiveEditCounter = 0;
+  ezHybridArray<const ezActionMap*, 3> mappings;
+  {
+    const ezActionMap* pCurrent = this;
+    while (pCurrent)
+    {
+      uiCurrentTransitiveEditCounter += pCurrent->m_uiEditCounter;
+      mappings.PushBack(pCurrent);
+      pCurrent = ezActionMapManager::GetActionMap(pCurrent->m_sParentMapping);
+    }
+  }
+
+  if (uiCurrentTransitiveEditCounter == m_uiTransitiveEditCounterOfRoot)
+  {
+    return &m_Root;
+  }
+  m_uiTransitiveEditCounterOfRoot = uiCurrentTransitiveEditCounter;
+
+  m_Root = TreeNode();
+  m_Descriptors.Clear();
+
+  for (ezInt32 i = (ezInt32)mappings.GetCount() - 1; i >= 0; --i)
```

Comment:
When I need to loop backwards with uint, I often do this:

```cpp
for (ezUInt32 iPlus1 = arr.GetCount(); iPlus1 > 0; --iPlus1)
{
  const ezUInt32 i = iPlus1 - 1;
}
```

Reply:
I would do that if I absolutely require an unsigned value. If not, I'd rather have one line less and the benefit of not accidentally picking the wrong counter variable in my code inside the loop.